code/controllers/subsystem/speech_controller.dm

```diff
@@ -0,0 +1,64 @@
+SUBSYSTEM_DEF(speech_controller)
+	name = "Speech Controller"
+	wait = 1
+	flags = SS_TICKER
+	priority = FIRE_PRIORITY_SPEECH_CONTROLLER//has to be high priority, second in priority ONLY to SSinput
+	init_order = INIT_ORDER_SPEECH_CONTROLLER
+	runlevels = RUNLEVELS_DEFAULT | RUNLEVEL_LOBBY
+
+	///associative list of the form: list(client mob = message or messages that mob is queued to say).
+	///this is our process queue, processed every tick.
+	var/list/queued_says_to_execute = list()
```

Comment:
does this need to be an associated list? could it be a list of `list(mob, messages)`?

Reply:
:+1:

Reply:
yes. now also includes any number of given arguments to put into the say() too since thats necessary for some procs